lib/weather/weather_report.ex

```diff
@@ -32,12 +31,9 @@ defmodule Weather.WeatherReport do
   end
 
   def handle_response({:ok, %HTTPoison.Response{ status_code: 404}}), do: { :error, "Not Found" }
-  def handle_response({:error, %HTTPoison.Error{reason: reason}}) do
-    IO.puts "#{reason}"
-  end
-  
+
   def parse_xml({xml, _}) do
-    Enum.into(@needed_fields, %{}, fn node_name -> extract_node(xml, node_name) end)
+    Enum.into(@needed_fields, %{}, &(extract_node(xml, &1)) )
```

Comment:
what did you end up thinking of this? do you prefer the shortcut syntax for anonymous functions or the longer form?